index.js

```diff
@@ -530,6 +521,77 @@ methods.concat('all').forEach(function(method){
   }
 })
 
+/**
+ * Process args for a call to use or error
+ *
+ * @param {array} args
+ * @private
+ */
+
+function processUseArgs (args) {
+  var offset = 0
+  var path = '/'
+
+  // default path to '/'
+  // disambiguate router.use([handler])
+  if (typeof args[0] !== 'function') {
+    var arg = args[0]
+
+    while (Array.isArray(arg) && arg.length !== 0) {
+      arg = arg[0]
+    }
+
+    // first arg is the path
+    if (typeof arg !== 'function') {
+      offset = 1
+      path = args[0]
+    }
+  }
+
+  return [path, flatten(slice.call(args, offset))]
+}
+
+/**
+ * Create layers for a call to use or error
+ *
+ * @param {string} path
+ * @param {array} callbacks
+ * @param {object} opts
+ * @private
+ */
+
+function createLayers (path, callbacks, opts) {
+  if (callbacks.length === 0) {
+    throw new TypeError('argument handler is required')
+  }
+
+  return callbacks.map(function createLayers (fn) {
+    if (typeof fn !== 'function') {
+      throw new TypeError('argument handler must be a function')
+    }
+
+    if ((!opts.error && !mwStack.isError(fn)) && fn.length === 4) {
+      // not a standard error handler
+      deprecate('registering error handlers with .use is deprecated, use .error or Stack.error instead')
+      opts.error = true
```

Comment:
I think this is an issue here, as it is modifying `opts` but that is an object used for all `callbacks`, yet the modification seems to be only for the given `callback` being mapped over at the time.

It would seem to reason that if `callbacks` was an array of "middleware", "error handler", "handler" then this would end up marking that last "hander" as an error handler incorrectly?

Reply:
Good catch!  I will write a test which covers this and then make sure that the `opts` are only modified locally.